core.lua

```diff
@@ -13,56 +17,75 @@ function BuffDuty:OnInitialize()
 end
 
 function BuffDuty:init()
-    self:RegisterChatCommand(CHAT_COMMAND, "Command")
+    self:RegisterChatCommand(DUTY_COMMAND, "CommandDuty")
     self:RegisterChatCommand(MESSAGE_COMMAND, "CommandMessage")
 
     --Init DataBase
     self.db = LibStub("AceDB-3.0"):New("BuffDutyDB", defaults)
     BuffDuty.Cache.duties_cache = self.db.factionrealm.duties_cache
     BuffDuty.Messages.custom_messages = self.db.factionrealm.custom_messages
 
+    BuffDuty.Cache:Initialise()
+
     BuffDuty.Messages:Initialise()
     BuffDuty.Messages:Load()
+
+    -- Version Update
+    if not (self.db.global.version.KEY == BuffDuty.VERSION.KEY) then
+        self.db.global.version = BuffDuty.VERSION -- Set updated version
+    end
 end
 
-local function executeLogic(input)
+local function executeDuty(input)
     -- Checks whether makes sense to assign people
-    if (GetNumGroupMembers() < 10) then
+    if (GetNumGroupMembers() < 10) then -- WOW API: https://wowwiki.fandom.com/wiki/API_GetNumGroupMembers
         BuffDuty.printInfoMessage("Current Group/Raid is too small. No sense in assigning buffs.")
         return
     end
 
-    -- Logic execution
-    local class, ch_type, channel_name, excluded, order = LibStub("AceConsole-3.0"):GetArgs(input, 5)
-    class, ch_type, channel_name = BuffDuty:convertArgs(class, ch_type, channel_name)
-    BuffDuty:validateArgs(class, ch_type, channel_name)
-    if ch_type ~= BuffDuty.CUSTOM_CHANNEL_TYPE then
-        order = excluded
-        excluded = channel_name
-    end
-    if excluded and string.sub(excluded, 1, 1) == "o" then
-        order = excluded
-    end
-    excluded = BuffDuty:convertPlayerList("e", excluded)
-    order = BuffDuty:convertPlayerList("o", order)
-    local duties = {}
-
-    if (BuffDuty.Cache:cacheContains(class, excluded) and next(BuffDuty.Cache:getFromCache(class, excluded))) then
-        duties = BuffDuty.Cache:getFromCache(class, excluded)
-    else
-        duties = BuffDuty:getDutiesTable(class, excluded, order)
-        BuffDuty.Cache:addToCache(class, excluded, duties)
-    end
+    local cmd = {
+        -- Base args, listed here for reference
+        class = nil,
+        channel_type = nil,
+        channel_id = nil,
+        -- Tables
+        excluded = {},
+        order = nil,
+        assign = nil,
+        -- Logic settings
+        own_group = {},
+        -- Custom message settings, listed here for reference
+        public_title = nil,
+        duty_line = nil,
+        duty_whisper = nil,
+        single_message = nil,
+        single_whisper = nil,
+        -- Flags
+        cache = true,
+        debug = false,
+    }
 
-    local cmd = {}
-    cmd.class = class
-    cmd.channel_name = channel_name
+    if BuffDuty.Console.parseDutyCommand(cmd, LibStub("AceConsole-3.0"):GetArgs(input, 20)) then
```

Comment:
Probably would be better to move `LibStub("AceConsole-3.0"):GetArgs(input, 20)` to a separate var.

Reply:
The issue is `GetArgs( )` returns multiple values.. putting it in a seperate var means using `pack()` and `unpack()` to pass it nicely to `parseCommand`... having it as an argument to the function call means it goes directly into the `...` parameter which is then easily convertable to a table using `{...}`

Reply:
I mean like that. Are you sure we will need to pack/unpack in such case?
```suggestion
    local args = LibStub("AceConsole-3.0"):GetArgs(input, 20)
    if BuffDuty.Console.parseDutyCommand(cmd, args) then
```

Reply:
Yes. 'args' in your example will only capture the first returned value (the rest are discarded)... you would need to use at least `local args = pack(LibStub("AceConsole-3.0"):GetArgs(input, 20))`

Reply:
Also turns out `pack` (or rather `table.pack`) wasn't introduced until Lua 5.2 ... I've defined a local function instead.